client/browser/src/shared/repo/backend.tsx

```diff
@@ -103,7 +103,10 @@ export function retryWhenCloneInProgressError<T>(): (v: Observable<T>) => Observ
         )
 }
 
-const trimRepoName = ({ repoName, ...rest }) => ({ ...rest, repoName: repoName.replace(/.git$/, '') })
+const trimRepoName = ({ repoName, ...rest }: { repoName: string }) => ({
```

Comment:
This will error when providing object literals and the `rest` is not clear in the return type.
```suggestion
const trimRepoName = <T extends { repoName: string }>({ repoName, ...rest }: T): T => ({
```